### PR TITLE
CacheKey for secret nonshareable keychain items on macOS

### PR DESCRIPTION
**Pull request by GTValentine**

### CODE REVIEW

File: IdentityCore/src/cache/key/MSIDMacKeychainNonshareableCredentialCacheKey.m [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    NSString* accessGroup = @"valentins_wonderful_access_group"; // TODO
+    NSString* appBundleId = [[NSBundle mainBundle] bundleIdentifier];
+    NSString* accountId = self.accountId;
+
+    return [NSString stringWithFormat:@"%@-%@-%@", accessGroup, appBundleId, accountId];
+}
+
+- (NSString *)service
+{
+    if(self.target.length > 0)
```
Comment (oldalton) [inline IdentityCore/src/cache/key/MSIDMacKeychainNonshareableCredentialCacheKey.m]: This check shouldn't be necessary, as we still keep delimiters even if target is missing. Please see implementation in MSIDDefaultCredentialCacheKey:

```
// kSecAttrService - (<credential_type>-<client_id>-<realm>-<enrollment_id>-<target>)
- (NSString *)serviceWithType:(MSIDCredentialType)type
                     clientID:(NSString *)clientId
                        realm:(NSString *)realm
                 enrollmentId:(NSString *)enrollmentId
                       target:(NSString *)target
                       appKey:(NSString *)appKey
{
    realm = realm.msidTrimmedString.lowercaseString;
    clientId = clientId.msidTrimmedString.lowercaseString;
    target = target.msidTrimmedString.lowercaseString;
    enrollmentId = enrollmentId.msidTrimmedString.lowercaseString;

    NSString *credentialId = [self credentialIdWithType:type clientId:clientId realm:realm enrollmentId:enrollmentId];
    NSString *service = [NSString stringWithFormat:@"%@%@%@",
                         credentialId,
                         keyDelimiter,
                         (target ? target : @"")];
    
    if (![NSString msidIsStringNilOrBlank:appKey])
    {
        service  = [NSString stringWithFormat:@"%@|%@", service, appKey];
    }
    
    return service;
}
```

For that matter, perhaps we could even reuse "appKey" used for broker for macOS and update macOS spec (let's discuss)

File: IdentityCore/src/cache/key/MSIDMacKeychainNonshareableCredentialCacheKey.m [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+- (NSData *)generic
+{
+    return [[self credentialId] dataUsingEncoding:NSUTF8StringEncoding];
+}
+
+- (NSString *)appKey
+{
+    return @"app_key"; //TODO
```
Comment (oldalton) [inline IdentityCore/src/cache/key/MSIDMacKeychainNonshareableCredentialCacheKey.m]: Why is this one overridden?

**Comment by GTValentine** [comment]
@d-col I looked at [the default key](https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/blob/dev/IdentityCore/src/cache/key/MSIDDefaultCredentialCacheKey.m), and it seems it already does everything it should. Could you please confirm?

**Comment by d-col** [comment]
@GTValentine You're right, using the default key is best, if possible.

**Pull request closed without merging**